src/dotnet-aspnet-codegenerator/Program.cs

```diff
@@ -6,127 +6,218 @@
+            var buildBasePath = app.Option("-b|--buildBasePath", "", CommandOptionType.SingleValue);
+            var dependencyCommand = app.Option("-nd|--no-dispatch", "", CommandOptionType.NoValue);
+
             app.OnExecute(() =>
             {
-                var serviceProvider = new ServiceProvider();
-                var context = CreateProjectContext(projectPath.Value());
-                
+                string project = projectPath.Value();
+                if (string.IsNullOrEmpty(project))
+                {
+                    project = Directory.GetCurrentDirectory();
+                }
                 var configuration = appConfiguration.Value() ?? "Debug";
-                if(configuration != null && !configuration.Equals("Release") && !configuration.Equals("Debug")) 
+                var projectFile = ProjectReader.GetProject(project);
+                var frameworksInProject = projectFile.GetTargetFrameworks().Select(f => f.FrameworkName);
+
+                var nugetFramework = FrameworkConstants.CommonFrameworks.NetCoreApp10;
+
+                if (!isProjectDependencyCommand)
                 {
-                    throw new ArgumentException($"Invalid value for configuration: {configuration}. {appConfiguration.Description}");
+                    // Invoke the tool from the project's build directory. 
+                    return BuildAndDispatchDependencyCommand(
+                        args, 
+                        frameworksInProject.FirstOrDefault(), 
+                        project, 
+                        buildBasePath.Value(), 
+                        configuration);
+                }
+                else
+                {
+                    if (!TryGetNugetFramework(framework.Value(), out nugetFramework))
+                    {
+                        throw new ArgumentException($"Could not understand the NuGetFramework information. Framework short folder name passed in was: {framework.Value()}");
+                    }
+
+                    var nearestNugetFramework = NuGetFrameworkUtility.GetNearest(
+                        frameworksInProject,
+                        nugetFramework,
+                        f => new NuGetFramework(f));
+
+                    if(nearestNugetFramework == null)
+                    {
+                        // This should never happen as long as we dispatch correctly.
+                        var msg = "Could not find a compatible framework to execute."
+                            + Environment.NewLine
+                            +$"Available frameworks in project:{string.Join($"{Environment.NewLine} -", frameworksInProject.Select(f => f.GetShortFolderName()))}";
+                        throw new InvalidOperationException(msg);
+                    }
+
+                    ProjectContext context = new ProjectContextBuilder()
```

Comment:
You no longer use this. The CodeGenCommand doesn't seem to either. I missing something?